File: app/src/main/java/fr/xgouchet/chronorg/data/models/Event.java
package fr.xgouchet.chronorg.data.models;

import android.os.Parcel;
import android.os.Parcelable;
import android.support.annotation.ColorInt;
import android.support.annotation.NonNull;

import org.joda.time.DateTime;
import org.joda.time.ReadableInstant;

/**
 * @author Xavier Gouchet
 */
public class Event implements Parcelable {

    private int id;
    private int projectId;
    @NonNull private String name = "";
    @NonNull private ReadableInstant instant = new DateTime("1970-01-01T00:00:00Z");
    @ColorInt private int color;

    public Event() {
    }


    public void setId(int id) {
        this.id = id;
    }

    public int getId() {
        return id;
    }

    public void setProjectId(int projectId) {
        this.projectId = projectId;
    }

    public int getProjectId() {
        return projectId;
    }

    public void setName(@NonNull String name) {
        this.name = name;
    }

    @NonNull public String getName() {
        return name;
    }

    public void setInstant(@NonNull ReadableInstant instant) {
        this.instant = instant;
    }

    public void setInstant(@NonNull String instant) {
        this.instant = new DateTime(instant);
    }

    @NonNull public ReadableInstant getInstant() {
        return instant;
    }

    public void setColor(@ColorInt int color) {
        this.color = color;
    }

    @ColorInt public int getColor() {
        return color;
    }


    @Override public int describeContents() {
        return 0;
    }

    @Override public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(this.id);
        dest.writeInt(this.projectId);
        dest.writeString(this.name);
        dest.writeString(this.instant.toString());
        dest.writeInt(this.color);
    }

    protected Event(Parcel in) {
        this.id = in.readInt();
        this.projectId = in.readInt();
        this.name = in.readString();
        this.instant = new DateTime(in.readString());
        this.color = in.readInt();
    }

    public static final Parcelable.Creator<Event> CREATOR = new Parcelable.Creator<Event>() {
        @Override public Event createFromParcel(Parcel source) {
            return new Event(source);
        }

        @Override public Event[] newArray(int size) {
            return new Event[size];
        }
    };
}

File: khronorg/src/main/java/fr/xgouchet/khronorg/feature/events/Event.kt
package fr.xgouchet.khronorg.feature.events

import android.graphics.Color
import android.os.Parcel
import android.os.Parcelable
import fr.xgouchet.khronorg.commons.time.getLocalTimeZone
import org.joda.time.DateTime
import org.joda.time.ReadableInstant

/**
 * @author Xavier F. Gouchet
 */
data class Event(var id: Int = -1,
                 var projectId: Int = -1,
                 var name: String = "",
                 var instant: ReadableInstant = DateTime("1984-04-20T18:00:00Z"),
                 var color: Int = Color.rgb(0xF6, 0x40, 0x2C))
    : Parcelable {

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel?, flags: Int) {
        dest?.let {
            it.writeInt(id)
            it.writeInt(projectId)
            it.writeString(name)
            it.writeString(instant.toString())
            it.writeInt(color)
        }
    }

    constructor(input: Parcel) : this() {
        id = input.readInt()
        projectId = input.readInt()
        name = input.readString()
        instant = DateTime(input.readString(), getLocalTimeZone())
        color = input.readInt()
    }

    companion object {
        @JvmField val CREATOR: Parcelable.Creator<Event> = object : Parcelable.Creator<Event> {
            override fun createFromParcel(source: Parcel): Event {
                return Event(source)
            }

            override fun newArray(size: Int): Array<Event?> {
                return arrayOfNulls(size)
            }
        }
    }
}